beancount/core/amount.py

```diff
@@ -48,6 +48,22 @@ def __add__(self,other):
     def __sub__(self,other):
         return sub(self,other)
 
+    def __mul__(self,other):
+        try:
+            other = Decimal(other)
+        except ValueError:
+            pass
+        return mul(self,other)
+
+    def __rmul__(self,other):
+        return self.__mul__(other)
```

Comment:
Just `__rmul__ = __mul__` is enough.

Reply:
Actually, without __rmul__ you can do "Amount * 4" but not "4 * Amount" .

Reply:
What I was trying to say is that you can replace the function definition with the assignment.

Reply:
Ok. Done.